spark/agent/src/main/scala/za/co/absa/spline/harvester/builder/WriteCommand.scala

```diff
@@ -14,10 +14,13 @@
  * limitations under the License.
  */
 
-package za.co.absa.spline.spark.adapter
+package za.co.absa.spline.harvester.builder
 
+import org.apache.spark.sql.SaveMode
 import org.apache.spark.sql.catalyst.plans.logical.{Command, LogicalPlan}
 
-abstract class AbstractWriteCommand extends Command {
-  val query: LogicalPlan
-}
+case class WriteCommand(
+  sourceIdentifier: SourceIdentifier,
+  mode: SaveMode,
```

Comment:
This is clearer but we have a boolean on the persistence model atm, it needs to be modified and few other services as well I think